app/services/courses_query.rb

```diff
@@ -49,6 +50,23 @@ def visa_sponsorship_scope
       )
   end
 
+  def study_modes_scope
+    study_modes = Array(params[:study_types]).compact_blank
```

Comment:
I'm not comfortable passing request params directly to a query object. We should be processing the params in the controller / mapping object.

Reply:
What is the benefit of processing the params at the controller level? As this would de-couple the param handling from the query object, is there a case where we'd like to pass through non-processed params?

Reply:
ActiveSupport::Parameters are view level objects. Query objects are database level objects. Mixing the two blurs those lines between the frontend and the backend. I like there to be clear distinction between the two.

Reply:
> It creates an implicit dependency between the Query object and the ActionController::Parameters class.

I'd disagree on the account of duck-typing. The underlying implementation at the Query Object level doesn't depend on the `params` being of type `ActionController::Parameters`, it only requires that the `params` respond to index accessors; `params[key]`.

Reply:
The hint is in the name `ActionController::Parameters`. I would like to keep ActionController constants in the request realm. If it's not related specifically to a HTTP request, I think it's better to not let these objects bleed into other domains